src/Administration/Resources/app/administration/src/module/sw-product/helper/sw-products-variants-generator.js

```diff
@@ -124,8 +126,21 @@ export default class VariantsGenerator extends EventEmitter {
                 numbers[variant.productNumber] = true;
             }
 
-            // Copy the hashed list with the sorted variations on the server.
-            let deleteQueue = deepCopyObject(hashed);
+            let deleteQueue = [];
+            if (!isAddOnly) {
+                // Copy the hashed list with the sorted variations on the server.
+                deleteQueue = deepCopyObject(hashed);
+            } else {
+                // Reassign product numbers
+                let increment = 1;
+                createQueue.forEach((variant) => {
```

Comment:
I tested this function, and it works fine. However, while adding Jest tests, I discovered that this specific code is never applied because variable `createQueue` is always empty  (as defined above `createQueue = []` ).

Could you please explain the purpose of this code? 
Will there be any negative impact if I delete it?
So far, I have tried deleting it, and everything seems to be working fine.

Reply:
cc @amenk

Reply:
Thank you very much for the review and catching this issue @nguyenky 

Some background: 

We ported the code originally from an internal module, and introduced this logic for the following reasons:

        /* The original function assigns the product number based on the fact that some variants might be deleted,
         * but we do not actually delete any so we have to take all variation on the server into account to avoid
         * duplicate product numbers.
         */

But as you say, it's currently not called. If all tests go well, let's remove it for good. I also did not see this problem again in testing - but please keep this case in mind when adding test cases - maybe some edge case is not covered yet in the ported code.